### DISCUSSION TITLE
STATE PROOF MIMIC ON REACH FRONTEND

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [reach-sh/reach-lang#1156] [General]
Hello Reach Fam,

Just wanted to ask if there is any methods, or implementation that uses the Algorand state proofs using the Reach standard library. There are many beneficial uses for Algorand state proofs but having to use the normal Algorand tooling can make it hard to build out an app that uses it as fast as possible. However, with Reach it could be easy, but sadly I do not know if Reach has something that works like those state proofs on the frontend(certainly not on the Smart Contract``.rsh`` side, but on the ``.mjs`` side). If so, I could use it to verify a transaction happened offchain in my app if I used a third-party service to relay that transaction.

Thanks.

**Comment by jeapostrophe** [comment]
We don't have any plans right now to add that. I also don't really understand what you're trying to enable with the state proofs... what do you imagine it would look like?

**Reply by kcelestinomaria** [reply]
I want to enable transactions move from two different networks so that my users can have the option to transfer tokens without the need of the first network. I am using a third-party server to host the second network. However, I still need to quickly verify from the network that the transaction happened when it was sent over second network without having to run a node. State proofs help with this and minimize trust. @jeapostrophe

**Reply by kcelestinomaria** [reply]
@jeapostrophe , I am also imagining state proofs come in handy when you want to send proof of tokens transferred that are not on a blockchain network but on a third party financial service

**Reply by jeapostrophe** [reply]
For your first thing, you'd need a light client, but there are no light clients yet.

For the second thing, the third-party service would need a light client also and a reliable way to get every state proof.

I don't think that state proofs are read for something user-facing like this yet and I don't think it is even an application-specific thing at all
